lte/gateway/c/session_manager/SessionCredit.cpp

```diff
@@ -304,6 +304,49 @@ SessionCredit::Usage SessionCredit::get_usage_for_reporting(
   return usage;
 }
 
+RequestedUnits SessionCredit::get_requested_credits_units() {
+  RequestedUnits requestedUnits;
+  uint64_t buckets_used_total =buckets_[USED_TX] + buckets_[USED_RX];
+
+  uint64_t total_requested = calculate_requested_unit(
+      received_granted_units_.total(),
+      ALLOWED_TOTAL, ALLOWED_FLOOR_TOTAL, buckets_used_total);
+
+  uint64_t tx_requested = calculate_requested_unit(
+      received_granted_units_.tx(),
+      ALLOWED_TX, ALLOWED_FLOOR_TX, buckets_[USED_TX]);
+
+  uint64_t rx_requested = calculate_requested_unit(
+      received_granted_units_.rx(),
+      ALLOWED_RX, ALLOWED_FLOOR_RX, buckets_[USED_RX]);
+
+  requestedUnits.set_total(total_requested);
+  requestedUnits.set_tx(tx_requested);
+  requestedUnits.set_rx(rx_requested);
+
+  return requestedUnits;
+}
+
+// returns either the last grant, or the difference between the last grant
+// and the credit remaining. Prevents over requesting in case we still have
+// credit available from the previous request
+uint64_t SessionCredit:: calculate_requested_unit(
+    CreditUnit cu, Bucket allowed,Bucket allowed_floor, uint64_t used){
+  if (cu.is_valid() == false){
+    return 0;
+  }
+  // get the current volume grant, or infer it in case of 0
+  int64_t grant = cu.volume()!=0 ? cu.volume() :
+                                   buckets_[allowed]-buckets_[allowed_floor];
+  int64_t remaining = buckets_[allowed] - used;
+  if (remaining >= 0 and grant >= remaining) {
```

Comment:
Woah is `and` a C++ thing???? That compiles? woa

Reply:
😅 😂. too many years working with python...

BTW, it looks like `and` is an alternative way for `&&` based on `SO 14882:2003 C++ standard 2.5/2`. 

I ll fix it


https://stackoverflow.com/questions/4251699/is-it-okay-to-use-and-or-etc-instead-of

Reply:
woah! learning something new everyday :D